cfme/common/vm_views.py

```diff
@@ -0,0 +1,366 @@
+
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class VMEntities(BaseEntitiesView):
+    """
+    Entities view for vms/instances collection destinations
+    """
+    adv_search_clear = Text('//div[@id="main-content"]//h1//span[@id="clear_search"]/a')
+
+
+class VMDetailsEntities(View):
+    """
+    Details entities view for vms/instances details destinations
+
+    VM's have 3-4 more tables, should inherit and add them there.
+    """
+    title = Text('//div[@id="main-content"]//h1//span[@id="explorer_title_text"]')
+    flash = FlashMessages('.//div[@id="flash_msg_div"]'
+                          '/div[@id="flash_text_div" or contains(@class, "flash_text_div")]')
+    properties = SummaryTable(title='Properties')
+    lifecycle = SummaryTable(title='Lifecycle')
+    relationships = SummaryTable(title='Relationships')
+    vmsafe = SummaryTable(title='VMsafe')
+    attributes = SummaryTable(title='Custom Attributes')  # Only displayed when assigned
+    compliance = SummaryTable(title='Compliance')
+    power_management = SummaryTable(title='Power Management')
+    security = SummaryTable(title='Security')
+    configuration = SummaryTable(title='Configuration')
+    diagnostics = SummaryTable(title='Diagnostics')
+    smart_management = SummaryTable(title='Smart Management')
+
+
+class ProvisionView(BaseLoggedInPage):
+    """
+    The provisioning view, with nested ProvisioningForm as `form` attribute.
+    Handles template selection before Provisioning form with `before_fill` method
+    """
+    title = Text('#explorer_title_text')
+
+    @View.nested
+    class form(View):  # noqa
+        """First page of provision form is image selection
+        Second page of form is tabbed with nested views
+        """
+        image_table = Table('//div[@id="pre_prov_div"]//table')
+        continue_button = Button('Continue')  # Continue button on 1st page, image selection
+        submit_button = Button('Submit')  # Submit for 2nd page, tabular form
+        cancel_button = Button('Cancel')
+
+        def before_fill(self, values):
+            # Provision from image is a two part form,
+            # this completes the image selection before the tabular parent form is filled
+            template_name = values.get('template_name',
+                                       self.parent_view.context['object'].template_name)
+            provider_name = self.parent_view.context['object'].provider.name
+            try:
+                row = self.image_table.row(name=template_name,
+                                           provider=provider_name)
+            except IndexError:
+                raise TemplateNotFound('Cannot find template "{}" for provider "{}"'
+                                       .format(template_name, provider_name))
+            row.click()
+            self.continue_button.click()
+            # TODO timing, wait_displayed is timing out and can't get it to stop in is_displayed
+            sleep(3)
```

Comment:
@psav IIRC it has to do with the fact that ProvisioningForm.is_displayed is false on the 1st part of the form, where only  `self.image_table` is there.  There may be an improvement in wait_displayed that would 'fix' this but didn't have time to dive into that in the scope of this PR.